modules/LemDCBM400600/main/lem_dcbm_400600_controller.cpp

```diff
@@ -197,8 +199,18 @@ void LemDCBM400600Controller::convert_livemeasure_to_powermeter(const std::strin
     current.DC = data.at("current");
     powermeter.current_A.emplace(current);
     powermeter.power_W.emplace(types::units::Power{data.at("power")});
-    powermeter.temperature1 = data.at("temperatureH");
-    powermeter.temperature2 = data.at("temperatureL");
+    if (!powermeter.temperatures.has_value()) {
+        powermeter.temperatures.emplace({types::temperature::Temperature{data.at("temperatureH"), "temperatureH"},
+                                         types::temperature::Temperature{data.at("temperatureL"), "temperatureL"}});
+    } else {
+        for (auto& sensor : *powermeter.temperatures) {
+            if (sensor.identification == "temperatureL") {
+                sensor.temperature = data.at("temperatureL");
+            } else if (sensor.identification == "temperatureH") {
+                sensor.temperature = data.at("temperatureH");
+            }
+        }
+    }
```

Comment:
the `powermeter` argument passed to this function is default initialized and the properties are then set in this function. Therefore this `else` block is never executed and can be dropped. Its also not required to check for `has_value()` in L202.

Apart from that I'd prefer a function signature like:
`types::powermeter::Powermeter LemDCBM400600Controller::convert_livemeasure_to_powermeter(const std::string& livemeasure)` that returns the powermeter object instead of passing it as an argument, but since this is legacy code im ok with keeping the signature like that

Reply:
But isn't powermeter.temperatures an optional vector so that check is still needed?

Reply:
maybe I miss something but afaics the `temperatures` property is always not set so the else block is never executed

Reply:
because the passed `powermeter` is default initialized and passed to this function

Reply:
ah you're right, then it's a good idea to drop the else block indeed!

Reply:
You're right, I didn't check where the powermeter value comes from. 
Since it is created by the calling function the else branch would never execute.
Thank you for catching that.
I made the changes, including the signature of the function since I am changing it again and the change is minor.